### PR TITLE
[CU-2yktmdy] Moved owner validation to call level

### PR DESCRIPTION
**Pull request by PoisonPhang**
## Issue
[[CU-2yktmdy]](https://app.clickup.com/t/2yktmdy)

## Description
Also deleted unused error for validating pool owners before adding to reward.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2yktmdy Move owner validation to pallet::call level in staking-rewards](https://app.clickup.com/t/2yktmdy)

File: code/parachain/frame/staking-rewards/src/lib.rs [hunk 1]
```diff
@@ -1281,71 +1306,62 @@ pub mod pallet {
...
-							from: from.clone(),
-							pool_id: *pool,
-							reward_currency,
-							reward_increment,
-						});
-						Ok(())
-					},
-					None => Err(Error::<T>::RewardsPoolNotFound.into()),
-				}
+			RewardPools::<T>::try_mutate(pool, |reward_pool| match reward_pool {
```
Comment (vimukthi-git) [inline code/parachain/frame/staking-rewards/src/lib.rs:1309]: Is there any change here other than the indentation?

Comment (PoisonPhang) [reply code/parachain/frame/staking-rewards/src/lib.rs:1309]: Just the removal of your TODO as discussed on slack.
Not sure why everything else got changed

Comment (vimukthi-git) [reply code/parachain/frame/staking-rewards/src/lib.rs:1309]: As long as you haven't made any change in the logic we are good.

**Review by vimukthi-git — APPROVED** [review]
LGTM

File: code/parachain/frame/staking-rewards/src/lib.rs [hunk 2]
```diff
@@ -521,10 +519,14 @@ pub mod pallet {
 			fnft_instance_id: T::FinancialNftInstanceId,
 			amount: T::Balance,
 		) -> DispatchResult {
-			let owner = ensure_signed(origin)?;
+			let who = Self::ensure_stake_owner(
```
Comment (vimukthi-git) [inline code/parachain/frame/staking-rewards/src/lib.rs:522]: Would be good to verify this in a small unit test for each action

Comment (PoisonPhang) [reply code/parachain/frame/staking-rewards/src/lib.rs:527]: 4158f3fe0dfcdeded697191ee0559424b790da10

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**